cdap-api/src/main/java/co/cask/cdap/api/Transactionals.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.data.DatasetContext;
+import org.apache.tephra.TransactionFailureException;
+
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Helper class for interacting with {@link Transactional}
+ */
+public final class Transactionals {
+
+  private Transactionals() {
+  }
+
+  /**
+   * Executes the given {@link TxRunnable} using the given {@link Transactional}.
+   *
+   * @param transactional the {@link Transactional} to use for transactional execution.
+   * @param runnable the {@link TxRunnable} to be executed inside a transaction
+   * @throws RuntimeException if failed to execute the given {@link TxRunnable} in a transaction.
+   * If the TransactionFailureException has a cause in it, the cause is propagated.
+   */
+  public static void execute(Transactional transactional, final TxRunnable runnable) {
+    try {
+      transactional.execute(new TxRunnable() {
+        @Override
+        public void run(DatasetContext context) throws Exception {
+          runnable.run(context);
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw propagate(e);
+    }
+  }
+
+  /**
+   * Executes the given {@link TxRunnable} using the given {@link Transactional}.
+   *
+   * @param transactional the {@link Transactional} to use for transactional execution.
+   * @param runnable the {@link TxRunnable} to be executed inside a transaction
+   * @param <X> exception type of propagate type
+   * @throws X if failed to execute the given {@link TxRunnable} in a transaction. If the TransactionFailureException
+   * has a cause in it, the cause is thrown as-is if it is an instance of X.
```

Comment:
Otherwise it will be wrapped with `RuntimeException` if it is not already a `RuntimeException`, right? We gotta document that as well.